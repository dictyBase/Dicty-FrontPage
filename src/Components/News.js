// @flow
import React from "react"
import { Flex, Box } from "grid-styled"
import twitterLogo from "./images/twitterLogo.png"
import FontAwsome from "react-fontawesome"


import {
  Container,
  Header,
  NewsBox,
  ListItems,
  NewsDate,
  NewsContent,
  NewsSource,
  SourceTitle,
  Link,
  NewsMore,
  NewsTitle,
  Img,
  Danger,
} from "./styles"

/**
 * The News component that will be displayed on the right in every
 * react web application of [dictyBase](http://dictybase.org).
 */

const News = (props: {
  /** List of news items */
  posts: Array<{
    "id": number,
    "date": number,
    "content": string,
    "source": string,
  }>,
}) => {
  const text = props.posts.map((post, index) =>
    <ListItems key={index}>
      <NewsDate>
        {post.date}
      </NewsDate>
      <NewsContent>
        <Danger dangerouslySetInnerHTML={{ __html: post.content }} />
      </NewsContent>
      <NewsSource>
        <SourceTitle>Source:</SourceTitle>
        {post.source}
      </NewsSource>
    </ListItems>,
  )

  return (
    <Container>
      <Header>
        <Flex wrap>
          <Box px={2} py={1} width={1 / 10}>
            <Link
              href="https://twitter.com/dictybase"
              alt="Dicty News at Twitter"
              target="new">
              <Img src={twitterLogo} />
            </Link>
          </Box>
          <Box px={2} py={1} width={9 / 10}>
            <FontAwsome name="globe fa-lg" />
            <NewsTitle>DICTY NEWS</NewsTitle>
          </Box>
        </Flex>
      </Header>
      <NewsBox>
        {text}
      </NewsBox>
      <NewsMore>
        <FontAwsome name="plus" />
        <Link href="" alt="more news">
          {" "}more news{" "}
        </Link>
      </NewsMore>
    </Container>
  )
}

export default News
