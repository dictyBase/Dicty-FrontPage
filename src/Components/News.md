{
  "src/Components/News.js": {
    "description": "The News component that will be displayed on the right in every\nreact web application of [dictyBase](http://dictybase.org).",
    "methods": [],
    "props": {
      "posts": {
        "flowType": {
          "name": "Array",
          "elements": [
            {
              "name": "signature",
              "type": "object",
              "raw": "{\n  \"id\": number,\n  \"date\": number,\n  \"content\": string,\n  \"source\": string,\n}",
              "signature": {
                "properties": [
                  {
                    "key": "id",
                    "value": {
                      "name": "number",
                      "required": true
                    }
                  },
                  {
                    "key": "date",
                    "value": {
                      "name": "number",
                      "required": true
                    }
                  },
                  {
                    "key": "content",
                    "value": {
                      "name": "string",
                      "required": true
                    }
                  },
                  {
                    "key": "source",
                    "value": {
                      "name": "string",
                      "required": true
                    }
                  }
                ]
              }
            }
          ],
          "raw": "Array<{\n  \"id\": number,\n  \"date\": number,\n  \"content\": string,\n  \"source\": string,\n}>"
        },
        "required": true,
        "description": "List of news items"
      }
    }
  },
  "src/Components/Slideshow.js": {
    "description": "",
    "methods": [
      {
        "name": "next",
        "docblock": null,
        "modifiers": [],
        "params": [],
        "returns": null
      },
      {
        "name": "previous",
        "docblock": null,
        "modifiers": [],
        "params": [],
        "returns": null
      },
      {
        "name": "renderImages",
        "docblock": null,
        "modifiers": [],
        "params": [],
        "returns": null
      },
      {
        "name": "renderDots",
        "docblock": null,
        "modifiers": [],
        "params": [],
        "returns": null
      }
    ]
  },
  "src/Components/styles/NewsContainer.js": {
    "description": "",
    "methods": []
  },
  "src/Components/styles/Danger.js": {
    "description": "",
    "methods": []
  },
  "src/Components/styles/Header.js": {
    "description": "",
    "methods": []
  },
  "src/Components/styles/Img.js": {
    "description": "",
    "methods": []
  },
  "src/Components/styles/Link.js": {
    "description": "",
    "methods": []
  },
  "src/Components/styles/ListItems.js": {
    "description": "",
    "methods": []
  },
  "src/Components/styles/NewsBox.js": {
    "description": "",
    "methods": []
  },
  "src/Components/styles/NewsContent.js": {
    "description": "",
    "methods": []
  },
  "src/Components/styles/NewsDate.js": {
    "description": "",
    "methods": []
  },
  "src/Components/styles/NewsMore.js": {
    "description": "",
    "methods": []
  },
  "src/Components/styles/NewsSource.js": {
    "description": "",
    "methods": []
  },
  "src/Components/styles/NewsTitle.js": {
    "description": "",
    "methods": []
  },
  "src/Components/styles/SourceTitle.js": {
    "description": "",
    "methods": []
  }
}