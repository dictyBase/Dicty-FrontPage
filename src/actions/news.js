// @flow
import { push } from "connected-react-router"
import {
  EDIT_NEWS,
  SAVE_NEWS_REQUEST,
  SAVE_NEWS_SUCCESS,
  SAVE_NEWS_FAILURE,
  FETCH_NEWS_REQUEST,
  FETCH_NEWS_SUCCESS,
  FETCH_NEWS_FAILURE,
} from "constants/types"
import { printError, createErrorObj } from "utils/actionHelpers"
import { fetchBySlugResource, fetchByIdResource } from "utils/fetchResources"

const server: string = process.env.REACT_APP_API_SERVER

const fetchNewsRequest = () => ({
  type: FETCH_NEWS_REQUEST,
  payload: {
    isFetching: true,
  },
})

const fetchNewsSuccess = (json: Object) => ({
  type: FETCH_NEWS_SUCCESS,
  payload: {
    isFetching: false,
    json,
  },
})

const fetchNewsFailure = error => ({
  type: FETCH_NEWS_FAILURE,
  payload: {
    error,
  },
})

const saveNewsRequest = () => ({
  type: SAVE_NEWS_REQUEST,
  payload: {
    isFetching: true,
  },
})

const saveNewsSuccess = () => ({
  type: SAVE_NEWS_SUCCESS,
  payload: {
    isFetching: false,
  },
})

const saveNewsFailure = error => ({
  type: SAVE_NEWS_FAILURE,
  payload: {
    error,
  },
})

// fetch news function that fetches data using async/await
// checks if header is correct, then either grabs data or displays error
export const fetchNews = (slug: string) => async (dispatch: Function) => {
  try {
    dispatch(fetchNewsRequest())
    const res = await fetch(`${fetchBySlugResource}/${slug}`)
    const contentType = res.headers.get("content-type")
    if (contentType && contentType.includes("application/vnd.api+json")) {
      const json = await res.json()
      if (res.ok) {
        dispatch(fetchNewsSuccess(json))
      } else {
        if (process.env.NODE_ENV !== "production") {
          printError(res, json)
        }

        dispatch(
          fetchNewsFailure(createErrorObj(res.status, json.errors[0].title)),
        )
        dispatch(push("/error"))
      }
    } else {
      dispatch(fetchNewsFailure(createErrorObj(res.status, res.statusText)))
      dispatch(push("/error"))
    }
  } catch (error) {
    dispatch(fetchNewsFailure(createErrorObj(error.name, error.message)))
    dispatch(push("/error"))
    if (process.env.NODE_ENV !== "production") {
      console.error(`Network error: ${error.message}`)
    }
  }
}

// template for future fetch request to get all news items
export const fetchAllNews = () => async (dispatch: Function) => {
  try {
    dispatch(fetchNewsRequest())
    const res = await fetch(`${fetchBySlugResource}/frontpage/news`)
    const contentType = res.headers.get("content-type")
    if (contentType && contentType.includes("application/vnd.api+json")) {
      const json = await res.json()
      if (res.ok) {
        dispatch(fetchNewsSuccess(json))
      } else {
        if (process.env.NODE_ENV !== "production") {
          printError(res, json)
        }

        dispatch(
          fetchNewsFailure(createErrorObj(res.status, json.errors[0].title)),
        )
        dispatch(push("/error"))
      }
    } else {
      dispatch(fetchNewsFailure(createErrorObj(res.status, res.statusText)))
      dispatch(push("/error"))
    }
  } catch (error) {
    dispatch(fetchNewsFailure(createErrorObj(error.name, error.message)))
    dispatch(push("/error"))
    if (process.env.NODE_ENV !== "production") {
      console.error(`Network error: ${error.message}`)
    }
  }
}

const doEdit = (content: Object) => ({
  type: EDIT_NEWS,
  payload: {
    content,
  },
})

export const editInline = (content: Object) => (dispatch: Function) => {
  dispatch(doEdit(content))
}

export const saveInlineEditing = (id: string, body: Object) => async (
  dispatch: Function,
  getState: Function,
) => {
  try {
    dispatch(saveNewsRequest())
    const res = await fetch(`${fetchByIdResource}/${id}`, {
      method: "PATCH",
      body: JSON.stringify(body),
      headers: {
        "Content-Type": "application/json",
        Authorization: `Bearer ${getState().auth.token}`,
      },
    })
    const contentType = res.headers.get("content-type")
    if (contentType && contentType.includes("application/vnd.api+json")) {
      const json = await res.json()
      if (res.ok) {
        dispatch(saveNewsSuccess())
      } else {
        if (process.env.NODE_ENV !== "production") {
          printError(res, json)
        }
        dispatch(
          saveNewsFailure(createErrorObj(res.status, json.errors[0].title)),
        )
        dispatch(push("/error"))
      }
    } else {
      dispatch(saveNewsFailure(createErrorObj(res.status, res.statusText)))
      dispatch(push("/error"))
    }
  } catch (error) {
    dispatch(saveNewsFailure(createErrorObj(error.name, error.message)))
    dispatch(push("/error"))
    if (process.env.NODE_ENV !== "production") {
      console.error(`Network error: ${error.message}`)
    }
  }
}

export const addNewsItem = (body: Object) => async (
  dispatch: Function,
  getState: Function,
) => {
  try {
    dispatch(saveNewsRequest())
    const res = await fetch(`${server}/contents`, {
      method: "POST",
      body: JSON.stringify(body),
      headers: {
        "Content-Type": "application/json",
        Authorization: `Bearer ${getState().auth.token}`,
      },
    })
    const contentType = res.headers.get("content-type")
    if (contentType && contentType.includes("application/vnd.api+json")) {
      const json = await res.json()
      if (res.ok) {
        dispatch(saveNewsSuccess())
        dispatch(push("/"))
      } else {
        if (process.env.NODE_ENV !== "production") {
          printError(res, json)
        }
        dispatch(
          saveNewsFailure(createErrorObj(res.status, json.errors[0].title)),
        )
        dispatch(push("/error"))
      }
    } else {
      dispatch(saveNewsFailure(createErrorObj(res.status, res.statusText)))
      dispatch(push("/error"))
    }
  } catch (error) {
    dispatch(saveNewsFailure(createErrorObj(error.name, error.message)))
    dispatch(push("/error"))
    if (process.env.NODE_ENV !== "production") {
      console.error(`Network error: ${error.message}`)
    }
  }
}

export const cancelEditing = () => (dispatch: Function) => {
  dispatch(push("/"))
}
