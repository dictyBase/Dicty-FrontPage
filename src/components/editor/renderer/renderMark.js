// @flow
import React, { type Node } from "react"
import { type Mark } from "slate"
import MARKS from "components/editor/constants/marks"

type Props = {
  children: Node,
  mark: Mark,
}

const renderMark = (props: Props) => {
  const { children, mark } = props
  switch (mark.type) {
    case MARKS.BOLD:
      return <strong>{children}</strong>

    case MARKS.ITALIC:
      return <em>{children}</em>

    case MARKS.UNDERLINE:
      return <u>{children}</u>

    case MARKS.STRIKETHROUGH:
      return <del>{children}</del>

    default:
      return null
  }
}

export default renderMark
