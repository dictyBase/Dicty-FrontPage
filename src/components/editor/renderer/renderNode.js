// @flow
import React, { type Node } from "react"
import { type Block } from "slate"
import styled from "styled-components"
import Video from "components/editor/renderer/Video"
import BLOCKS from "components/editor/constants/blocks"
import INLINES from "components/editor/constants/inlines"

type Props = {
  node: Block,
  children: Node,
  attributes: Object,
}

const Image = styled.img`
  display: block;
  max-width: 100%;
  max-height: 20em;
  box-shadow: ${props => (props.selected ? "0 0 0 2px blue;" : "none")};
`

const renderNode = (props: Props) => {
  const { attributes, children, node } = props
  switch (node.type) {
    case BLOCKS.UL_LIST:
      return <ul {...attributes}>{children}</ul>

    case BLOCKS.OL_LIST:
      return <ol {...attributes}>{children}</ol>

    case BLOCKS.LIST_ITEM:
      return <li {...attributes}>{children}</li>

    case BLOCKS.HEADING_1:
      return <h1 {...attributes}>{children}</h1>

    case BLOCKS.HEADING_2:
      return <h2 {...attributes}>{children}</h2>

    case BLOCKS.HEADING_3:
      return <h3 {...attributes}>{children}</h3>

    case BLOCKS.BLOCKQUOTE:
      return <blockquote {...attributes}>{children}</blockquote>

    case INLINES.LINK: {
      const { data } = node
      const href = data.get("href")
      return (
        <a {...attributes} href={href}>
          {children}
        </a>
      )
    }

    case BLOCKS.IMAGE: {
      const { data, isSelected } = node
      const src = data.get("src")
      const className = isSelected ? "selected" : "unselected"
      const style = { display: "inline-block" }
      return (
        <img
          src={src}
          className={className}
          {...attributes}
          style={style}
          alt=""
        />
      )
    }

    case BLOCKS.ALIGN_LEFT:
      return (
        <div style={{ textAlign: "left" }} {...attributes}>
          {children}
        </div>
      )

    case BLOCKS.ALIGN_CENTER:
      return (
        <div style={{ textAlign: "center" }} {...attributes}>
          {children}
        </div>
      )

    case BLOCKS.ALIGN_RIGHT:
      return (
        <div style={{ textAlign: "right" }} {...attributes}>
          {children}
        </div>
      )

    case BLOCKS.HR:
      return <hr {...attributes} />

    case BLOCKS.TABLE:
      return (
        <table {...attributes}>
          <tbody>{children}</tbody>
        </table>
      )

    case BLOCKS.TABLE_ROW:
      return <tr {...attributes}>{children}</tr>

    case BLOCKS.TABLE_CELL:
      return <td {...attributes}>{children}</td>

    case BLOCKS.VIDEO:
      return <Video {...props} />

    default:
      return null
  }
}

export default renderNode
