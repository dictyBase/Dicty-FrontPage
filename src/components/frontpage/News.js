// @flow
import React from "react"
import { Link as RouterLink } from "react-router-dom"
import { Flex, Box } from "rebass"
import styled from "styled-components"
import FontAwesome from "react-fontawesome"
import NewsList from "./NewsList"
import twitterLogo from "images/twitterLogo.png"

import {
  NewsContainer,
  Header,
  ListBox,
  Link,
  MoreLink,
  NewsStockTitle,
  Img,
} from "styles"

const StyledRouterLink = styled(RouterLink)`
  text-decoration: none;
`

/** Widget that displays the most recent Dicty news */

const News = () => {
  return (
    <NewsContainer>
      <Header>
        <Flex wrap>
          <Box px={2} py={1} width={1 / 10}>
            <Link
              href="https://twitter.com/dictybase"
              alt="Dicty News at Twitter"
              target="new">
              <Img src={twitterLogo} />
            </Link>
          </Box>
          <Box px={2} py={1} width={9 / 10}>
            <FontAwesome name="globe fa-lg" />
            <NewsStockTitle>DICTY NEWS</NewsStockTitle>
          </Box>
        </Flex>
      </Header>
      <ListBox>
        <NewsList />
      </ListBox>
      <MoreLink>
        <FontAwesome name="plus" />
        <StyledRouterLink to="/news" alt="more news">
          {" "}
          more news{" "}
        </StyledRouterLink>
      </MoreLink>
    </NewsContainer>
  )
}

export default News
