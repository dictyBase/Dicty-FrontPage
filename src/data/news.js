module.exports = [
  {
    date: "2015.08.19",
    source: "Dicty Stock Center",
    content:
      "Why axenic strains grow by fluid uptake: We have added the axeB gene to Chromosome 3 at the locus which is largely deleted in axenic strains and thus responsible for their ability to grow axenically. Only an axeB 3' fragment could be generated on the AX4 genome assembly. The full length gene in non-axenic strains such as DdB codes for the conserved neurofibromin 1, a RasGAP, which has been shown to be involved in macropinocytosis and phagocytosis. For more information, see Bloomfield et al.",
    //"<strong>Why axenic strains grow by fluid uptake</strong>: We have added the <a href='http://dictybase.org/gene/DDB_G0350652' target='new'> axeB gene </a> to Chromosome 3 at the locus which is largely deleted in axenic strains and thus responsible for their ability to grow axenically. Only an axeB 3' fragment could be generated on the AX4 genome assembly. The full length gene in non-axenic strains such as DdB codes for the conserved neurofibromin 1, a RasGAP, which has been shown to be involved in macropinocytosis and phagocytosis. For more information, see <a href='http://dictybase.org/publication/pubmed/25815683' target='new'>Bloomfield et al</a>.",
  },
  {
    date: "2015.08.12",
    source: "National Geographic",
    content:
      "Dicty in the News: The National Geographic series a Not Exactly Rocket Science has published a nice piece about the latest results reporting how Burkholderia turns Dicty amooeba into farmers,published by Suzanne di Salvo and colleagues from the Strassmann/Queller lab.",
    //"Dicty in the News: The National Geographic series a <a href='http://phenomena.nationalgeographic.com/2015/08/24/the-bacteria-that-turn-amoebas-into-farmers/?sf12476278=1' target='new'>Not Exactly Rocket Science</a> has published a nice piece about the latest results reporting how Burkholderia turns Dicty amooeba into farmers, <a href='http://www.pnas.org/content/early/2015/08/18/1511878112' target='new'>published</a> by Suzanne di Salvo and colleagues from the Strassmann/Queller lab.",
  },
  {
    date: "2015.08.07",
    source: "Dicty Stock Center",
    content:
      "Starting for DSC orders placed after Monday August 10, 2015 (12:00 AM CDT), Dicty stocks will incur charges. Please see the complete price and payment information on the Order page",
  },
  {
    date: "2015.08.06",
    source: "Dicty Stock Center",
    content:
      "The bug in the DSC ordering system as reported on July 31, has been fixed",
  },
  {
    date: "2015.07.20",
    source: "Dicty Stock Center",
    content:
      "Alert Dicty Stock Center: We currently have problems receiving email notifications of Stock Center orders. If you have ordered in recent months and have not received your order within a week to 10 days, please email Kerry Sheppard and forward your order. If you currently place an order, please also forward your automatic confirmation to Kerry. We are working on the problem and will let you know when fixed. Thank you for your patience and help.",
  },
  {
    date: "2015.07.23",
    source: "dictyBase",
    content:
      "Dicty in the News: The recent discovery of the protein Callipygian in Dictyostelium discoideum provides insights into how round, symmetrical cells become polarized: News Medical with comments from Kristen Swaney and Peter Devreotes.",
  },
  {
    date: "2015.06.12",
    source: "dictyBase",
    content:
      "Dicty in the News: Press releases from three recent research publications highlight the use of Dictyostelium: electrotaxis in Futurity by Zhao and Devreotes; kin recognition in Yahoo News by Ho and Shaulsky; cheating in EurekAlert by Ostrowski, Strassmann and Queller.",
  },
  {
    date: "2015.05.13",
    source: "dictyBase",
    content:
      "Bill Loomis has written a memoir called My Life with Dicty and makes it available on dictyBase.",
  },
  {
    date: "2015.03.11",
    source: "dicty15",
    content:
      "Dicty 2015. Registration is now open for the Annual International Dictyostelium Conference.",
  },
  {
    date: "2015.03.02",
    source: "dictyBase",
    content:
      "Dicty in the News: A recent paper by Falk Hilmann and Thomas Winckler is discussed in DocCheck News in Germany.",
  },
  {
    date: "2014.11.19",
    source: "dictyBase",
    content:
      "The Dicty Stock Center will be closed for Thanksgiving in the week of November 24th - 28th. Please be advised that orders placed from now until November 30 will be shipped in early December. Please do not submit any materials during that week. Thank you for your understanding.",
  },
  {
    date: "2014.11.06",
    source: "Petra Fey",
    content:
      "Dicty in the News: Dictyostelium research details how a cell's gradient sensing and cytoskeletal architecture dictate one mode of chemotactic movement: The Scientist with comments from Chuan-Hsiang Huang",
  },
  {
    date: "2014.10.17",
    source: "DictyBase",
    content:
      "Dicty in the News: Research on the Dictyostelium lpten gene potentially leads to a new way to treat cancer: Drug Discovery and Development with comments from David Soll",
  },
]
