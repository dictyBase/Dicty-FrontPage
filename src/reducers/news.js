// @flow
import {
  EDIT_NEWS,
  SAVE_NEWS_REQUEST,
  SAVE_NEWS_SUCCESS,
  SAVE_NEWS_FAILURE,
  FETCH_NEWS_REQUEST,
  FETCH_NEWS_SUCCESS,
  FETCH_NEWS_FAILURE,
} from "constants/types"

const initialState = {
  content: null,
  isFetching: false,
  error: null,
}

const newsReducer = (state: Object = initialState, action: Object) => {
  switch (action.type) {
    case FETCH_NEWS_REQUEST:
      return {
        ...state,
        isFetching: true,
      }
    case FETCH_NEWS_SUCCESS:
      const slugName = action.payload.json.data.attributes.slug
      return {
        ...state,
        isFetching: false,
        [slugName]: action.payload.json,
      }
    case FETCH_NEWS_FAILURE:
      return {
        ...state,
        isFetching: false,
        error: action.payload.error,
      }
    case EDIT_NEWS:
      return {
        ...state,
        content: action.payload.content,
      }
    case SAVE_NEWS_REQUEST:
      return {
        ...state,
        isFetching: true,
      }
    case SAVE_NEWS_SUCCESS:
      return {
        ...state,
        isFetching: false,
      }
    case SAVE_NEWS_FAILURE:
      return {
        ...state,
        isFetching: false,
        error: action.payload.error,
      }
    default:
      return state
  }
}

export default newsReducer
