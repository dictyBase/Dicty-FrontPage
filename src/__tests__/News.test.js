import React from "react"
import News from "../components/frontpage/News"
import renderer from "react-test-renderer"
import news from "../././data/news"

test("News snapshot test", () => {
  const component = renderer.create(<News posts={news} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
