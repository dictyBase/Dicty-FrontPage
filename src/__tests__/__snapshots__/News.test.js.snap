// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News snapshot test 1`] = `
<div
  className="sc-bxivhb jucAju"
>
  <div
    className="sc-htoDjs jZuqdH"
  >
    <div
      className="sc-htpNat jtWIOA sc-bdVaJa iHZvIS"
      is={null}
    >
      <div
        className="sc-bdVaJa hZXWJA"
        is={null}
      >
        <a
          alt="Dicty News at Twitter"
          className="sc-bZQynM guiyJe"
          href="https://twitter.com/dictybase"
          target="new"
        >
          <img
            className="sc-chPdSV bsCNOM"
            src="twitterLogo.png"
          />
        </a>
      </div>
      <div
        className="sc-bdVaJa iFPAFX"
        is={null}
      >
        <span
          aria-hidden={true}
          className="fa fa-globe fa-lg"
        />
        <span
          className="sc-kAzzGY dfXKJP"
        >
          DICTY NEWS
        </span>
      </div>
    </div>
  </div>
  <ul
    className="sc-iwsKbI ZIKrF"
  >
    <li
      className="sc-gqjmRU bimEIq"
    >
      <span
        className="sc-jTzLTM iFsAid"
      >
        2015.08.19
      </span>
      <span
        className="sc-jzJRlG gcDScp"
      >
        <span
          className="sc-kGXeez lcKMhk"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<strong>Why axenic strains grow by fluid uptake</strong>: We have added the <a href='http://dictybase.org/gene/DDB_G0350652' target='new'> axeB gene </a> to Chromosome 3 at the locus which is largely deleted in axenic strains and thus responsible for their ability to grow axenically. Only an axeB 3' fragment could be generated on the AX4 genome assembly. The full length gene in non-axenic strains such as DdB codes for the conserved neurofibromin 1, a RasGAP, which has been shown to be involved in macropinocytosis and phagocytosis. For more information, see <a href='http://dictybase.org/publication/pubmed/25815683' target='new'>Bloomfield et al</a>.",
            }
          }
        />
      </span>
      <span
        className="sc-fjdhpX iwGYHP"
      >
        <span
          className="sc-cSHVUG gUqckI"
        >
          Source:
        </span>
        Dicty Stock Center
      </span>
    </li>
    <li
      className="sc-gqjmRU bimEIq"
    >
      <span
        className="sc-jTzLTM iFsAid"
      >
        2015.08.12
      </span>
      <span
        className="sc-jzJRlG gcDScp"
      >
        <span
          className="sc-kGXeez lcKMhk"
          dangerouslySetInnerHTML={
            Object {
              "__html": "Dicty in the News: The National Geographics series a <a href='http://phenomena.nationalgeographic.com/2015/08/24/the-bacteria-that-turn-amoebas-into-farmers/?sf12476278=1' target='new'>Not Exactly Rocket Science</a> has published a nice piece about the latest results reporting how Burkholderia turns Dicty amooeba into farmers, <a href='http://www.pnas.org/content/early/2015/08/18/1511878112' target='new'>published</a> by Suzanne di Salvo and colleagues from the Strassmann/Queller lab.",
            }
          }
        />
      </span>
      <span
        className="sc-fjdhpX iwGYHP"
      >
        <span
          className="sc-cSHVUG gUqckI"
        >
          Source:
        </span>
        National Geographics
      </span>
    </li>
    <li
      className="sc-gqjmRU bimEIq"
    >
      <span
        className="sc-jTzLTM iFsAid"
      >
        2015.08.07
      </span>
      <span
        className="sc-jzJRlG gcDScp"
      >
        <span
          className="sc-kGXeez lcKMhk"
          dangerouslySetInnerHTML={
            Object {
              "__html": "Starting for DSC orders placed after Monday August 10, 2015 (12:00 AM CDT), Dicty stocks will incur charges. Please see the complete price and payment information on the Order page",
            }
          }
        />
      </span>
      <span
        className="sc-fjdhpX iwGYHP"
      >
        <span
          className="sc-cSHVUG gUqckI"
        >
          Source:
        </span>
        Dicty Stock Center
      </span>
    </li>
    <li
      className="sc-gqjmRU bimEIq"
    >
      <span
        className="sc-jTzLTM iFsAid"
      >
        2015.08.06
      </span>
      <span
        className="sc-jzJRlG gcDScp"
      >
        <span
          className="sc-kGXeez lcKMhk"
          dangerouslySetInnerHTML={
            Object {
              "__html": "The bug in the DSC ordering system as reported on July 31, has been fixed",
            }
          }
        />
      </span>
      <span
        className="sc-fjdhpX iwGYHP"
      >
        <span
          className="sc-cSHVUG gUqckI"
        >
          Source:
        </span>
        Dicty Stock Center
      </span>
    </li>
    <li
      className="sc-gqjmRU bimEIq"
    >
      <span
        className="sc-jTzLTM iFsAid"
      >
        2015.07.20
      </span>
      <span
        className="sc-jzJRlG gcDScp"
      >
        <span
          className="sc-kGXeez lcKMhk"
          dangerouslySetInnerHTML={
            Object {
              "__html": "Alert Dicty Stock Center: We currently have problems receiving email notifications of Stock Center orders. If you have ordered in recent months and have not received your order within a week to 10 days, please email Kerry Sheppard and forward your order. If you currently place an order, please also forward your automatic confirmation to Kerry. We are working on the problem and will let you know when fixed. Thank you for your patience and help.",
            }
          }
        />
      </span>
      <span
        className="sc-fjdhpX iwGYHP"
      >
        <span
          className="sc-cSHVUG gUqckI"
        >
          Source:
        </span>
        Dicty Stock Center
      </span>
    </li>
    <li
      className="sc-gqjmRU bimEIq"
    >
      <span
        className="sc-jTzLTM iFsAid"
      >
        2015.07.23
      </span>
      <span
        className="sc-jzJRlG gcDScp"
      >
        <span
          className="sc-kGXeez lcKMhk"
          dangerouslySetInnerHTML={
            Object {
              "__html": "Dicty in the News: The recent discovery of the protein Callipygian in Dictyostelium discoideum provides insights into how round, symmetrical cells become polarized: News Medical with comments from Kristen Swaney and Peter Devreotes.",
            }
          }
        />
      </span>
      <span
        className="sc-fjdhpX iwGYHP"
      >
        <span
          className="sc-cSHVUG gUqckI"
        >
          Source:
        </span>
        dictyBase
      </span>
    </li>
    <li
      className="sc-gqjmRU bimEIq"
    >
      <span
        className="sc-jTzLTM iFsAid"
      >
        2015.06.12
      </span>
      <span
        className="sc-jzJRlG gcDScp"
      >
        <span
          className="sc-kGXeez lcKMhk"
          dangerouslySetInnerHTML={
            Object {
              "__html": "Dicty in the News: Press releases from three recent research publications highlight the use of Dictyostelium: electrotaxis in Futurity by Zhao and Devreotes; kin recognition in Yahoo News by Ho and Shaulsky; cheating in EurekAlert by Ostrowski, Strassmann and Queller.",
            }
          }
        />
      </span>
      <span
        className="sc-fjdhpX iwGYHP"
      >
        <span
          className="sc-cSHVUG gUqckI"
        >
          Source:
        </span>
        dictyBase
      </span>
    </li>
    <li
      className="sc-gqjmRU bimEIq"
    >
      <span
        className="sc-jTzLTM iFsAid"
      >
        2015.05.13
      </span>
      <span
        className="sc-jzJRlG gcDScp"
      >
        <span
          className="sc-kGXeez lcKMhk"
          dangerouslySetInnerHTML={
            Object {
              "__html": "Bill Loomis has written a memoir called My Life with Dicty and makes it available on dictyBase.",
            }
          }
        />
      </span>
      <span
        className="sc-fjdhpX iwGYHP"
      >
        <span
          className="sc-cSHVUG gUqckI"
        >
          Source:
        </span>
        dictyBase
      </span>
    </li>
    <li
      className="sc-gqjmRU bimEIq"
    >
      <span
        className="sc-jTzLTM iFsAid"
      >
        2015.03.11
      </span>
      <span
        className="sc-jzJRlG gcDScp"
      >
        <span
          className="sc-kGXeez lcKMhk"
          dangerouslySetInnerHTML={
            Object {
              "__html": "Dicty 2015. Registration is now open for the Annual International Dictyostelium Conference.",
            }
          }
        />
      </span>
      <span
        className="sc-fjdhpX iwGYHP"
      >
        <span
          className="sc-cSHVUG gUqckI"
        >
          Source:
        </span>
        dicty15
      </span>
    </li>
    <li
      className="sc-gqjmRU bimEIq"
    >
      <span
        className="sc-jTzLTM iFsAid"
      >
        2015.03.02
      </span>
      <span
        className="sc-jzJRlG gcDScp"
      >
        <span
          className="sc-kGXeez lcKMhk"
          dangerouslySetInnerHTML={
            Object {
              "__html": "Dicty in the News: A recent paper by Falk Hilmann and Thomas Winckler is discussed in DocCheck News in Germany.",
            }
          }
        />
      </span>
      <span
        className="sc-fjdhpX iwGYHP"
      >
        <span
          className="sc-cSHVUG gUqckI"
        >
          Source:
        </span>
        dictyBase
      </span>
    </li>
    <li
      className="sc-gqjmRU bimEIq"
    >
      <span
        className="sc-jTzLTM iFsAid"
      >
        2014.11.19
      </span>
      <span
        className="sc-jzJRlG gcDScp"
      >
        <span
          className="sc-kGXeez lcKMhk"
          dangerouslySetInnerHTML={
            Object {
              "__html": "The Dicty Stock Center will be closed for Thanksgiving in the week of November 24th - 28th. Please be advised that orders placed from now until November 30 will be shipped in early December. Please do not submit any materials during that week. Thank you for your understanding.",
            }
          }
        />
      </span>
      <span
        className="sc-fjdhpX iwGYHP"
      >
        <span
          className="sc-cSHVUG gUqckI"
        >
          Source:
        </span>
        dictyBase
      </span>
    </li>
    <li
      className="sc-gqjmRU bimEIq"
    >
      <span
        className="sc-jTzLTM iFsAid"
      >
        2014.11.06
      </span>
      <span
        className="sc-jzJRlG gcDScp"
      >
        <span
          className="sc-kGXeez lcKMhk"
          dangerouslySetInnerHTML={
            Object {
              "__html": "Dicty in the News: Dictyostelium research details how a cell's gradient sensing and cytoskeletal architecture dictate one mode of chemotactic movement: The Scientist with comments from Chuan-Hsiang Huang",
            }
          }
        />
      </span>
      <span
        className="sc-fjdhpX iwGYHP"
      >
        <span
          className="sc-cSHVUG gUqckI"
        >
          Source:
        </span>
        Petra Fey
      </span>
    </li>
    <li
      className="sc-gqjmRU bimEIq"
    >
      <span
        className="sc-jTzLTM iFsAid"
      >
        2014.10.17
      </span>
      <span
        className="sc-jzJRlG gcDScp"
      >
        <span
          className="sc-kGXeez lcKMhk"
          dangerouslySetInnerHTML={
            Object {
              "__html": "Dicty in the News: Research on the Dictyostelium lpten gene potentially leads to a new way to treat cancer: Drug Discovery and Development with comments from David Soll",
            }
          }
        />
      </span>
      <span
        className="sc-fjdhpX iwGYHP"
      >
        <span
          className="sc-cSHVUG gUqckI"
        >
          Source:
        </span>
        DictyBase
      </span>
    </li>
  </ul>
  <div
    className="sc-VigVT cGFCFH"
  >
    <span
      aria-hidden={true}
      className="fa fa-plus"
    />
    <a
      alt="more news"
      className="sc-bZQynM guiyJe"
      href=""
    >
       
      more news
       
    </a>
  </div>
</div>
`;
